### ISSUE TITLE
Add joints for 3d physics

### ISSUE THREAD

**Issue by d954mas** [defold/defold#9656]
**Is your feature request related to a problem? Please describe (REQUIRED):**
I need to make door is 3d game. So i need hinge constraint from bullet.

But now defold only support joint for 2d physics.

dmLogError("joints are currently only available in 2D physics");

**Describe the solution you'd like (REQUIRED):**
hinge constraint for 3d physics.

**Describe alternatives you've considered (REQUIRED):**
I think i can make door with set rotation. But it will be much harder.

**Additional context (OPTIONAL):**

![2024-10-25_22-06-42](https://github.com/user-attachments/assets/dd22cdfb-bf32-4f15-a105-11b88c7623ce)

**Comment by d954mas** [comment]
I made doors for my game without joint. So this features is not blocking me.

But joints for 3d is  still cool feature that needed in 3d engine:)

**Comment by aglitchman** [comment]
Can't agree more!

**Comment by JCash** [comment]
The current idea is to add a `bullet3d` namespace (analoguous to the `box2d` namespace), where these functions can be added without api restrictions.

**Comment by britzl** [comment]
> [@JCash](https://github.com/JCash) has it been determined if the `bullet3d` namespace will have identical or near-identical functions as `b2d.body`? Or more?

No, not necessarily. Since it is a namespace specifically for `bullet3d` it should be designed in a way which makes sense for the `bullet3d` API. It should probably match the C++ API fairly closely.

> If there is a design-doc for the `bullet3d` namespace

There is no design doc (yet!).

**Comment by Wolfe2x7** [comment]
> No, not necessarily. Since it is a namespace specifically for bullet3d it should be designed in a way which makes sense for the bullet3d API. It should probably match the C++ API fairly closely.

I understand. The gist was whether or not a user (like me) would request a function that might otherwise be left on the table, since avoiding adding any more features than what is considered necessary is the Defold way. 😉

Asking while the iron is hot -- or before it is heated -- seemed good, since you asked about my top 3 physics requests.

Don't worry, I don't want `bullet3d`'s flawed vehicle module. 😆
